### ISSUE TITLE
Homebridge-TCC reads status but does not respond

### ISSUE THREAD

**Issue by thevoyager3** [NorthernMan54/homebridge-tcc#56]
Hi northernman, thank you for you TCC plug-in. After installing it with MKH1 I have it reading the current temperature and status of my AC but any attempt to change the AC status leads to a Not Responding message. On the homebridge Log I get the following message: “ Error: This callback function has already been called by someone else; it can only be called one time.” Any ideas? Thank you.

**Comment by NorthernMan54** [comment]
Can you share more of the log around the error, they typically have details around where the error occurred in the code.

**Comment by mendeslr** [comment]
Thanks for replying so quickly. The log is bellow:

`[2020-4-18 23:24:50] [Thermostat] Change LivingRoom1 {}
 has already been called by someone else; it can only be called one time.
    at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/once.js:12:13
    at /usr/local/lib/node_modules/homebridge-tcc/index.js:228:11
    at _fulfilled (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:854:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:883:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:816:13)
    at /usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:624:44
    at runSingle (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:137:13)
    at flush (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:125:13)
    at process._tickCallback (internal/process/next_tick.js:112:11)

[2020-4-18 23:24:21] [Thermostat] Change LivingRoom1 {}
[2020-4-18 23:24:49] [Thermostat] updateValues 1
[2020-4-18 23:24:50] [Thermostat] Change LivingRoom1 {}`

**Comment by thevoyager3** [comment]
Thanks for replaying so quickly. The log is bellow:

> [2020-4-18 23:24:50] [Thermostat] Change LivingRoom1 {}
> has already been called by someone else; it can only be called one time.
> at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/once.js:12:13
> at /usr/local/lib/node_modules/homebridge-tcc/index.js:228:11
> at _fulfilled (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:854:54)
> at self.promiseDispatch.done (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:883:30)
> at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:816:13)
> at /usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:624:44
> at runSingle (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:137:13)
> at flush (/usr/local/lib/node_modules/homebridge-tcc/node_modules/q/q.js:125:13)
> at process._tickCallback (internal/process/next_tick.js:112:11)
> 
> [2020-4-18 23:24:21] [Thermostat] Change LivingRoom1 {}
> [2020-4-18 23:24:49] [Thermostat] updateValues 1
> [2020-4-18 23:24:50] [Thermostat] Change LivingRoom1 {}

**Comment by thevoyager3** [comment]
And I also got this following erros (always when changing the thermostat temperature):

> [2020-4-18 23:29:04] [Thermostat] setHeatCoolSetpoint LivingRoom1 0 0
> Error  302
> [2020-4-18 23:29:04] [Thermostat] Error: setHeatCoolSetpoint LivingRoom1 HTTP Error 
>

**Comment by NorthernMan54** [comment]
For the first issue, please update and try again.  For the second issue do you have any logs or more details?

**Comment by thevoyager3** [comment]
Hi Northernman, I apologize for not getting back to your. I realized that my homebridge and it’s plug ins where not updating properly and ended up erasing the whole thing. This time I installed hoobs (really wanted to try that out) and the plug in is running smoother now. However it is still very unresponsive (but to be honest, all the TCC controls are even in their own app). I’ll keep testing and if a specific bug comes out I will let u know. Thanks for working in this plug in.
